bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -822,10 +831,10 @@ public void run()
       };
     }
 
-    protected void release(boolean wait)
+    protected void release(boolean wait, boolean writer)
     {
       final int refCount = this.refCount.decrementAndGet();
-      if (refCount == 0 && storage != null) {
+      if (refCount == 0 && storage != null && canRelease(writer)) {
```

Comment:
Functionally the responsibility of the method is to check if the block can be released (the refCount check) and release it so why not here. Additionally, the release would be triggered by the subscriber not the publisher.